src/targets/maven.ts

```diff
@@ -81,10 +81,24 @@ export class MavenTarget extends BaseTarget {
    * @returns the maven config for this target.
    */
   private getMavenConfig(): MavenTargetConfig {
+    const targetSecrets = this.getTargetSecrets();
+    const outerTargetSettings = this.getOuterTargetSettings();
+
+    if (this.config.android) {
+      const androidSettings = this.getAndroidSettings();
+      return {
+        ...targetSecrets,
+        ...outerTargetSettings,
+        ...androidSettings,
+      };
+    }
+
+    this.logger.warn(
+      'You may need the Android configuration and it was not found in the configuration file.'
+    );
```

Comment:
not sure if this should be a `warn` level since the `android` config is optional.

Reply:
I find anything below a `warn` to not be enough when an `android` release may file because the config is missing.

Reply:
Then this should be `info` or even less and then we should have a hard-fail when this is required but missing.